pkg/vpc/vpc.go

```diff
@@ -276,3 +368,38 @@ func cleanupSubnets(spec *api.ClusterConfig) {
 		}
 	}
 }
+
+func validatePublicSubnet(subnets []*ec2.Subnet) error {
+	for _, sn := range subnets {
+		if sn.MapPublicIpOnLaunch == nil || !*sn.MapPublicIpOnLaunch {
+			return fmt.Errorf("found mis-configured subnet %q. Expected public subnet with property "+
+				"\"MapPublicIpOnLaunch\" enabled. Without it new nodes won't get an IP assigned", *sn.SubnetId)
```

Comment:
Should we collect all public subnets that have `MapPublicIpOnLaunch` disabled and show them in the error message?

Reply:
good idea